### ISSUE TITLE
ClickHouse dialect: Add support for ALTER TABLE UPDATE/DELETE mutations

### ISSUE THREAD

**Issue by adfineberg** [sqlfluff/sqlfluff#7434]
## Description

The ClickHouse dialect doesn't currently support `ALTER TABLE ... UPDATE` and `ALTER TABLE ... DELETE` mutation syntax, which is ClickHouse's way of performing data modifications.

## Example SQL

```sql
-- UPDATE mutation
ALTER TABLE my_table
    UPDATE column_name = 'new_value'
    WHERE condition = true
    SETTINGS mutations_sync = 1;

-- DELETE mutation  
ALTER TABLE my_table
    DELETE WHERE condition = true
    SETTINGS mutations_sync = 1;
```

## Current Behavior

```
L:   4 | P:   1 |  PRS | Line 4, Position 1: Found unparsable section: 'ALTER TABLE
                       | my_table\n    U...'
WARNING: Parsing errors found and dialect is set to 'clickhouse'.
```

## Expected Behavior

These statements should parse successfully as valid ClickHouse syntax.

## Reference

- [ClickHouse UPDATE mutations docs](https://clickhouse.com/docs/en/sql-reference/statements/alter/update)
- [ClickHouse DELETE mutations docs](https://clickhouse.com/docs/en/sql-reference/statements/alter/delete)

## Notes

ClickHouse uses `ALTER TABLE ... UPDATE/DELETE` instead of standard SQL `UPDATE/DELETE` statements because mutations work differently in ClickHouse's columnar architecture - they rewrite data parts asynchronously rather than modifying rows in place.

The grammar would need to be added to `AlterTableStatementSegment` in `dialect_clickhouse.py`, something like:

```python
# UPDATE mutation
Sequence(
    "UPDATE",
    Delimited(
        Sequence(
            Ref("ColumnReferenceSegment"),
            Ref("EqualsSegment"),
            Ref("ExpressionSegment"),
        ),
    ),
    Ref("WhereClauseSegment"),
    Ref("SettingsClauseSegment", optional=True),
),
# DELETE mutation
Sequence(
    "DELETE",
    Ref("WhereClauseSegment"),
    Ref("SettingsClauseSegment", optional=True),
),
```

## Version

sqlfluff 4.0.0
